### PR TITLE
fix: Add missing RBAC permissions for finalizers

### PR DESCRIPTION
**Pull request by adwk67**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by adwk67** [comment]
Test still fails for Hostname.
`kubectl get listener/listener-nginx-long-name-approaching-k8s-limits-0 -o jsonpath='http://{.status.ingressAddresses[0].address}:{.status.ingressAddresses[0].ports.http}/pod-name'`
returns
`http://worker-wfcawizc:31352/pod-name1000800000`
and not 
`listener-nginx-long-name-approaching-k8s-limits-0`

**Comment by adwk67** [comment]
Test case for Hostname will be omitted on OKD:
Kind: (the test runs but the curl commands are deactivated)
```
--- PASS: kuttl (20.60s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-false_addressType-Hostname (10.30s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-false_addressType-IP (10.31s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-false_addressType-HostnameConservative (10.25s)
```
OKD: 
```
--- PASS: kuttl (120.01s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-true_addressType-Hostname (40.39s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-true_addressType-HostnameConservative (101.85s)
        --- PASS: kuttl/harness/smoke-nodeport_openshift-true_addressType-IP (111.76s)
```

File: deploy/helm/listener-operator/templates/roles.yaml [hunk 1]
```diff
@@ -76,9 +76,14 @@ rules:
       - ""
     resources:
       - nodes
+      - persistentvolumes
```
Comment (sbernauer) [inline deploy/helm/listener-operator/templates/roles.yaml:79]: For me the permissions seem a bit excessive. As you have access to OS, I trust you to make sure we need all of them :)

**Review by sbernauer — APPROVED** [review]

**Review by sbernauer — APPROVED** [review]
Thanks!

**Pull request merged**